### PR TITLE
Small soulbound improvement

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
Very quick change to add a `slimefun:soulbound` key. Also added a utility method to add the lore alongside the key in SlimefunUtils.

## Changes
Added `slimefun:soulbound` key, SlimefunUtils.setSoulbound(ItemStack) and made SoulboundRune use this. (Couldn't find other uses of "Soulbound", others uses can be added later).

## Related Issues
N/A

## Checklist

- [X] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [X] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [X] I followed the existing code standards and didn't mess up the formatting.
- [X] I did my best to add documentation to any public classes or methods I added.

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java [hunk 1]
```diff
@@ -104,6 +113,23 @@ else if (item.hasItemMeta()) {
...
     }
 
+    /**
+     * Set an {@link ItemStack} to be Soulbound. This will add the {@link #SOULBOUND_LORE} and 
+     * add a {@link NamespacedKey} to the item so it can be quickly identified by {@link #isSoulbound(ItemStack)}.
+     * 
+     * @param item The {@link ItemStack} you want to make Soulbound.
+     * @see #isSoulbound(ItemStack)
+     */
+    public static void setSoulbound(@Nonnull ItemStack item) {
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java:123]: Could probably use a "boolean soulbound" argument to toggle it.

File: src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java [hunk 2]
```diff
@@ -104,6 +113,23 @@ else if (item.hasItemMeta()) {
...
+        final ItemMeta meta = item.getItemMeta();
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/utils/SlimefunUtils.java:124]: Should probably also do a check whether it is already soulbound or not.

**Comment by TheBusyBiscuit** [comment]
Build failed because you forgot to update your code to the method you changed.

**Comment by TheBusyBiscuit** [comment]
Also merge conflicts now XD

**Comment by WalshyDev** [comment]
I'm smart and now reee give me a min, on another branch

**Pull request merged**